autorest/adal/token.go

```diff
@@ -264,41 +259,57 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 	)
 }
 
-// NewServicePrincipalTokenFromMSI creates a ServicePrincipalToken via the MSI VM Extension.
-func NewServicePrincipalTokenFromMSI(oauthConfig OAuthConfig, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
-	return newServicePrincipalTokenFromMSI(oauthConfig, resource, managedIdentitySettingsPath, callbacks...)
+// Resource enumerates resources from which MSI authentication is supported
+type Resource string
+
+const (
+	// VirtualMachine specifies MSI support for VMs
+	VirtualMachine Resource = "VM"
+)
+
+// GetMSIEndpoint gets the correct endpoint for MSI auth depending on which resource
+// code is running on.
+func GetMSIEndpoint(resource Resource) (string, error) {
```

Comment:
Do we need to leave some space for expansion here? What if some of our future helper methods need information about the resource itself to get the MSI endpoint? Maybe I'm crazy, convince me that this isn't the case :)

Reply:
Example?

Reply:
Chatting offline we have concluded it is a better idea to have a func for each resource.